ngraph_onnx/onnx_importer/ops_bridge/opset_01.py

```diff
@@ -571,6 +571,49 @@ def Conv(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> Ngra
     return make_convolution_op(onnx_node, ng_inputs)
 
 
+def ConvTranspose(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> NgraphNode
+    """Calculate convolution transpose."""
+    if len(ng_inputs) == 3:
+        data, weights, bias = ng_inputs
+    elif len(ng_inputs) == 2:
+        data, weights = ng_inputs
+        bias = ng.constant(0, dtype=get_dtype(data.get_element_type()))
+
+    strides = get_strides(onnx_node)
+    dilation = get_dilations(onnx_node)
+    padding_below, padding_above = get_pads(onnx_node)
+
+    output_padding = onnx_node.get_attribute_value('output_padding')
+
+    weights_shape = list(weights.shape)
+    data_shape = list(data.shape)
+
+    num_spatial_dims = len(data.shape) - 2
+
+    data_dilation_strides = [1, 1]
+
+    data_batch_shape = [1] * (num_spatial_dims + 2)
+    data_batch_shape[0] = data_shape[0]
+    data_batch_shape[1] = weights_shape[1]
+
+    for i in range(num_spatial_dims):
+        data_batch_shape[i + 2] = ((padding_below[i] + ((data_shape[i + 2] - 1) * strides[i] + 1) + output_padding[i]) -
```

Comment:
Here is some magic. Please try adding some comment explaining the mathematical formula, or maybe some urls what is mapped into what.